addons/mail/models/mail_template.py

```diff
@@ -131,19 +131,36 @@ def _compute_template_category(self):
 
     @api.model
     def _search_template_category(self, operator, value):
-        if operator in ['in', 'not in'] and isinstance(value, list):
-            value_templates = self.env['mail.template'].search([]).filtered(
-                lambda t: t.template_category in value
-            )
-            return [('id', operator, value_templates.ids)]
+        imd_res_ids = self.browse(self.env['ir.model.data'].search([
+            ('model', '=', 'mail.template'), ('module', '!=', '__export__')
+        ]).mapped('res_id'))
```

Comment:
Should probably sudo for the ir.model.data read

Also more intuitive name maybe
```suggestion
        internal_templates = self.browse(self.env['ir.model.data'].search([
            ('model', '=', 'mail.template'), ('module', '!=', '__export__')
        ]).mapped('res_id'))
```